base/iddict.jl

```diff
@@ -153,16 +171,24 @@ end
 
 length(d::IdDict) = d.count
 
+isempty(d::IdDict) = length(d) == 0
+
 copy(d::IdDict) = typeof(d)(d)
 
 function get!(d::IdDict{K,V}, @nospecialize(key), @nospecialize(default)) where {K, V}
-    val = ccall(:jl_eqtable_get, Any, (Any, Any, Any), d.ht, key, secret_table_token)
-    if val === secret_table_token
+    keyindex = ht_keyindex2!(d, key)
+
+    if keyindex < 0
+        # If convert call fails we need the key to be deleted
+        @inbounds d[-keyindex] = nothing
```

Comment:
```suggestion
```

Reply:
Does the deleted state not require they key to be `nothing`? Though this prevents a segfault so I'm probably missing something.

Reply:
I think I get it now.

Reply:
Nevermind, the segfault was because I meant to write `d.ht[-keyindex] = nothing`. My concern is if the `convert` fails we have a valid key inserted and the value is NULL. This will prevent us from ever updating the value as we will hit this assert https://github.com/JuliaLang/julia/blob/e22db88127197cb3236e350cdb4ccbfff6cde232/src/iddict.c#L63-L69

I can move the key updating code out of `ht_keyindex2!` if that helps.